src/ck-core/ckcheckpoint.C

```diff
@@ -214,14 +220,22 @@ void CkCheckpointMgr::Checkpoint(const char *dirname, CkCallback cb, bool _reque
 	requestStatus = _requestStatus;
 	// make dir on all PEs in case it is a local directory
 	CmiMkdir(dirname);
-	bool success = true;
-        if (CmiNumPartitions() > 1) {
-          ostringstream partDir;
-          partDir << dirname;
-          addPartitionDirectory(partDir);
-          CmiMkdir(partDir.str().c_str());
-        }
 
+	// Create partition directories (if applicable)
+	ostringstream dirPath;
+	dirPath << dirname;
+	if (CmiNumPartitions() > 1) {
+		addPartitionDirectory(dirPath);
+		CmiMkdir(dirPath.str().c_str());
+	}
+
+	// Create subdirectories by grouping PEs
```

Comment:
I'd add some more descriptive comments about why this change was made to the code, something like: 
```suggestion
	// Due to file system issues we have observed, divide checkpoints
	// into subdirectories to avoid having too many files in a single directory 
```